### PR TITLE
Add Support for RP2.65-F

### PR DESCRIPTION
**Pull request by BigTuna94**
The RP2.65-F is my custom 65% mechanical keyboard based on the RP2040 MCU.
The '-F' signifies "Faders", which are analog linear potentiometers build onto the board.

Open source HW repo: https://github.com/BigTuna94/RP2.65-F

Since KMK did not have support for analog potentiometers, I added a module modeled after the encoder module.

### CODE REVIEW

**Comment by kdb424** [comment]
I'm generally a fan of this. I'll allow for some time for other reviewers to comment, but once tests are fixed, I won't block merge.

**Review by kdb424 — APPROVED** [review]

**Comment by BigTuna94** [comment]
@xs5871 I'm always happy to receive feedback! I actually agree about the separate base/interface/handler. I'd be happy to distill the potentiometer module down to fewer classes (and remove any mention of GPIO).

As far as the consumer control, I hadn't realized that there was a method to send control codes directly... seems that `keyboard._hid_helper.hid_send()` is the method that I need..

**Comment by BigTuna94** [comment]
@xs5871 I've done some revisions based on your feedback. I think it still makes sense to have a separate PotentiometerHandler class, since it makes instantiating multiple potentiometers a little easier from the user keymap perspective.

Thank you for advising me to use the kmk HID CC from the core - this setup is actually noticeably faster than the adafruit library!

File: kmk/modules/potentiometer.py
Comment (xs5871) [inline kmk/modules/potentiometer.py:23]: In my opinion the return type should be a proper class `PotentiometerState` and not a dictionary.
The encoder module also uses a dict, but I think a class is more pythonic.

Comment (xs5871) [inline kmk/modules/potentiometer.py:23]: In my opinion the return type should be a proper class `PotentiometerState` and not a dictionary.
The encoder module also uses a dict, but I think a class is more pythonic.

**Comment by xs5871** [comment]
Nothing left to reasonably complain about. Ready to merge from my point of view.

**Pull request merged**
